atomix/cluster/src/main/java/io/atomix/raft/partition/impl/RaftPartitionServer.java

```diff
@@ -101,46 +100,45 @@ public RaftPartitionServer(
 
   @Override
   public CompletableFuture<RaftPartitionServer> start() {
-    final RaftStartupMetrics raftStartupMetrics = new RaftStartupMetrics(partition.name());
-    final long bootstrapStartTime;
-    log.info("Starting server for partition {}", partition.id());
-    final long startTime = System.currentTimeMillis();
-    final CompletableFuture<RaftServer> serverOpenFuture;
-    if (partition.members().contains(localMemberId)) {
-      if (server != null && server.isRunning()) {
-        return CompletableFuture.completedFuture(null);
-      }
-      synchronized (this) {
-        try {
+    try {
+      final RaftStartupMetrics raftStartupMetrics = new RaftStartupMetrics(partition.name());
+      final long bootstrapStartTime;
+      log.info("Starting server for partition {}", partition.id());
+      final long startTime = System.currentTimeMillis();
+      final CompletableFuture<RaftServer> serverOpenFuture;
+      if (partition.members().contains(localMemberId)) {
+        if (server != null && server.isRunning()) {
+          return CompletableFuture.completedFuture(null);
+        }
+        synchronized (this) {
           initServer();
```

Comment:
Caused by: java.lang.IllegalStateException: Expected to find a snapshot at index >= log's first index 213419494, but found snapshot 213252958. A previous snapshot is most likely corrupted.
		at io.atomix.raft.impl.RaftContext.verifySnapshotLogConsistent(RaftContext.java:226) ~[zeebe-atomix-cluster-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
		at io.atomix.raft.impl.RaftContext.<init>(RaftContext.java:203) ~[zeebe-atomix-cluster-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
		at io.atomix.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:258) ~[zeebe-atomix-cluster-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
		at io.atomix.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:232) ~[zeebe-atomix-cluster-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
		at io.atomix.raft.partition.impl.RaftPartitionServer.buildServer(RaftPartitionServer.java:190) ~[zeebe-atomix-cluster-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
		at io.atomix.raft.partition.impl.RaftPartitionServer.initServer(RaftPartitionServer.java:156) ~[zeebe-atomix-cluster-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
```

I went for wrapping all invocations into a try block to make sure a future is always returned